fedscale/utils/models/tensorflow_model_provider.py

```diff
@@ -0,0 +1,64 @@
+import tensorflow as tf
+
+
+class RowCount(tf.keras.metrics.Metric):
+    def __init__(self, name='row_count', **kwargs):
+        super(RowCount, self).__init__(**kwargs)
+        self.count = self.add_weight('count', initializer='zeros')
+
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        self.count.assign_add(tf.reduce_sum(tf.cast(tf.shape(y_true)[0], self.dtype)))
+
+    def reset_state(self):
+        self.count.assign(0)
+
+    def result(self):
+        return self.count
+
+
+def build_resnet50(args):
+    model = tf.keras.applications.resnet.ResNet50(
+        include_top=True,
+        weights=None,
+        input_tensor=None,
+        input_shape=[32, 32, 3],
+        pooling=None,
+        classes=args.num_classes
+    )
+    optimizer = tf.keras.optimizers.SGD(learning_rate=args.learning_rate, momentum=0.9,
+                                        nesterov=False, name='SGD')
+    model.compile(optimizer=optimizer, loss='categorical_crossentropy',
+                  metrics=['accuracy', RowCount()])
+    return model
+
+
+def build_mobilenet_v3_small(args):
+    model = tf.keras.applications.MobileNetV3Small(
+        input_shape=[32, 32, 3],
```

Comment:
Hardcoding input size here seems to cause mismatch problems when dataset size varies